packages/flutter_tools/lib/src/doctor.dart

```diff
@@ -403,13 +404,16 @@ class IntelliJValidatorOnMac extends IntelliJValidator {
     }
 
     try {
-      for (FileSystemEntity dir in fs.directory('/Applications').listSync()) {
-        if (dir is Directory) {
-          checkForIntelliJ(dir);
-          if (!dir.path.endsWith('.app')) {
-            for (FileSystemEntity subdir in dir.listSync()) {
-              if (subdir is Directory)
-                checkForIntelliJ(subdir);
+      var checkDirectories = [ '/Applications' , path.join(homeDirPath, 'Applications') ];
```

Comment:
Style nits: Use explicit type annotations and generics, no spaces immediately before or after the opening/closing bracket:
`List<String> installPaths = <String> ['/Applications',...];`